assets/large_language_models/rag/components/src/flow_creation.py

```diff
@@ -131,6 +138,80 @@ def get_deployment_and_model_name(s):
     return (deployment_name, model_name)
 
 
+def get_user_alias_from_credential(credential):
+    """Get alias from credential. Copied from PF code."""
+    import jwt
+    token = credential.get_token("https://storage.azure.com/.default").token
+    decode_json = jwt.decode(token, options={"verify_signature": False, "verify_aud": False})
+    try:
+        email = decode_json["upn"]
+        return email.split("@")[0]
+    except Exception:
+        # use oid when failed to get upn, e.g. service principal
+        return decode_json["oid"]
+
+
+class CustomFileStorageClient(FileStorageClient):
+    """Wrapper around FileStorageClient to allof for custom directory client."""
+
+    def __init__(self, credential: str, file_share_name: str, account_url: str, azure_cred):
+        """Initialize client."""
+        super().__init__(credential=credential, file_share_name=file_share_name, account_url=account_url)
+        try:
+            user_alias = get_user_alias_from_credential(azure_cred)
+        except Exception:
+            # fall back to unknown user when failed to get credential.
+            user_alias = "unknown_user"
+        self._user_alias = user_alias
+
+        # TODO: update this after we finalize the design for flow file storage client
+        # create user folder if not exist
+        for directory_path in ["Users", f"Users/{user_alias}", f"Users/{user_alias}/Promptflows"]:
+            self.directory_client = ShareDirectoryClient(
+                account_url=account_url,
+                credential=credential,
+                share_name=file_share_name,
+                directory_path=directory_path,
+            )
+
+            # try to create user folder if not exist
+            try:
+                self.directory_client.create_directory()
+            except ResourceExistsError:
+                pass
+
+
+def upload_code_files(ws, name):
+    """Upload the files in the code flow directory."""
+    try:
+        credential = DefaultAzureCredential()
+        # Check if given credential can get token successfully.
+        credential.get_token("https://management.azure.com/.default")
+    except Exception:
+        # Fall back to InteractiveBrowserCredential in case DefaultAzureCredential not work
```

Comment:
this will not work for jobs and will cause the job to hang.

Reply:
Can you check how files are uploaded from components in mlindex creation or data gen etc.

Reply:
bummer. I will look for an alternative

Reply:
I have changed this, please take another look